doc/error-handling.md

```diff
@@ -0,0 +1,324 @@
+```
+std::expected<SomeType, Error>::func(Arg arg) {
+    int errorCode = checkArg(arg);
+    if(isNoError(errorCode)) {
+        SomeType result = computeResult(arg);
+        // optionally do something with result
+        return result;
+    }
+    return Error(errorCode);
+}
+```
+The caller is responsible for handling (or propagating) the error.
+
+```
+auto result = func(arg);
+if(result.has_error()) {
+    auto error = result.error();
+    //handle or propagate the error
+} else {
+    auto value = result.value();
+    //proceed by using the value
+}
+```
+
+Alternatively a functional approach can be used.
+
+```
+auto successFunc = [](SomeType& result) { 
+    auto value = result.value();
+    //proceed by using the value
+};
+
+auto errorFunc = [](Error& result) { 
+    auto error = result.error();
+    //handle the error
+};
+
+func(arg).and_then(successFunc).or_else(errorFunc);
+```
+
+# Open Points
+
+## Centralized Error Handling
+
+It may be desirable to have centralized error handling instance where runtime errors on application side are logged and (maybe) handled.
+This could also be done in RouDi (by sending information to RouDi), but RouDi already has too much responsibility. Preferably this should be done by a separate application with this sole purpose.
+If the application cannot reach the central handler, it shall try to handle the error locally if possible (at least log it).
+
+However, it might be too slow if this would rely on error transmission and responses. If this is to be implemented, the exact mechanism has to be decided on.
+
+## 3rd Party Error Handling
+We need to decide how to provide an interface for 3rd party error handling, especially for the runtime. This interface will probably rely on hooks/callbacks. The signature and callsites of these needs to be discussed.
+This is related to centralized error handling as well.
+
+## Overriding Specific Error Reaction
+* Do we want to provide the ability to override error reaction based on e.g. error codes?
+* Do we want to disable certain error levels? (if so, this should preferably happen at compile 
+time with no or few runtime artifacts). It could be an option to e.g. disable all MODERATE error reaction at compile time.
+* It is probably not reasonable to allow disabling reaction on FATAL errors.
+
+This is also related to the hooks for 3rd party error handling we may want to provide.
+
+## Return in Case of Fatal Error
+The reporting code does not need to be able to continue properly in case of a FATAL error, but there needs to be a return after the error handler call. While the error handler is not *required* to return, it still might under certain circumstances (e.g. a mock error handler in a test case). 
+
+The (complete) intended behavior of the error handler requires some further clarification, especially in the case of FATAL errors. In the case of non-FATAL errors the code invoking the error handler must be able to continue after the error handler returns.
```

Comment:
Add comma `In the case of non-FATAL errors, `

Reply:
No :-)